### ISSUE TITLE
Seasonal transform

### ISSUE THREAD

**Issue by fgregg** [observablehq/plot#2137]
I find myself making a lot of cycle / season line charts like https://observablehq.com/@fgregg/chicago-citykeys-issued

Right now, i have to do some data munging to do them, and it would be really great if it was a supported transform. 

perhaps relates to #133

**Comment by Fil** [comment]
Yes it would be useful to have as a `cycle: "1 year"` scale option, maybe?
Related experiment: https://observablehq.com/@observablehq/plot-cyclical-time-scale

**Comment by fgregg** [comment]
wow, that's such a nice solution @Fil!

**Comment by fgregg** [comment]
a `cycle` option would be great, maybe with a `cycle_start: 2024-06-01` option as well?
